### PR TITLE
README fixes and consistency improvements

### PR DESCRIPTION
**Pull request by guillaumekln**
This PR contains a few fixes and harmonizes some names and section titles. For example:

* TF Addons, TF-Addons, TFA, etc. -> TensorFlow Addons
* TF -> TensorFlow
* MacOS -> macOS
* PyPi -> PyPI
* custom-ops -> custom ops

Let me know what you think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,36 +40,36 @@ developments that cannot be integrated into core TensorFlow
...
 import tensorflow as tf
 import tensorflow_addons as tfa
 ```
 
 ### Python Op Compatility
-TensorFlow Addons is actively working towards forward compatiblity with TF2.x. 
-However there are still a few private API uses within the respository so at the moment 
-we can only gurentee compatibility with the versions of TF which it was tested against. 
-Warnings will be emitted during TFA import if your TensorFlow version does not match 
+TensorFlow Addons is actively working towards forward compatiblity with TensorFlow 2.x.
```
Comment (Squadrick) [inline README.md:65]: Typo: compatiblity -> compatibility

File: README.md [hunk 2]
```diff
@@ -40,36 +40,36 @@ developments that cannot be integrated into core TensorFlow
...
 import tensorflow_addons as tfa
 ```
 
 ### Python Op Compatility
-TensorFlow Addons is actively working towards forward compatiblity with TF2.x. 
-However there are still a few private API uses within the respository so at the moment 
-we can only gurentee compatibility with the versions of TF which it was tested against. 
-Warnings will be emitted during TFA import if your TensorFlow version does not match 
+TensorFlow Addons is actively working towards forward compatiblity with TensorFlow 2.x. 
+However, there are still a few private API uses within the respository so at the moment
```
Comment (Squadrick) [inline README.md:66]: Typo: respository -> repository

File: README.md [hunk 3]
```diff
@@ -79,22 +79,22 @@ what it was tested against.
...
+TensorFlow C++ APIs are not stable and thus we can only guarantee compatibility with the 
+version TensorFlow Addons was built against. It is possible custom ops will work with multiple 
+versions of TensorFlow, but there is also a chance for segmentation faults or other problematic crashes.
 Warnings will be emitted when loading a custom op if your TensorFlow version does not match 
 what it was built against.
 
-Additionally, registering custom-ops does not have an ABI stable interface so it is 
+Additionally, custom ops registration does not have a stable ABI interface so it is 
 required that users have a compatible installation of TensorFlow even if the versions 
 match what we had build against. A simplification of this is that **TensorFlow Addons
```
Comment (Squadrick) [inline README.md:90]: Correction: had build -> had built.

**Comment by guillaumekln** [comment]
Thanks @Squadrick, I fixed those as well.

**Pull request merged**